ouroboros-consensus-cardano/src/byron/Ouroboros/Consensus/Byron/Ledger/HeaderValidation.hs

```diff
@@ -56,9 +59,13 @@ instance BasicEnvelopeValidation ByronBlock where
 instance ValidateEnvelope ByronBlock where
   type OtherHeaderEnvelopeError ByronBlock = ByronOtherHeaderEnvelopeError
 
-  additionalEnvelopeChecks cfg _ledgerView hdr =
+  additionalEnvelopeChecks cfg _ledgerView hdr = do
       when (fromIsEBB newIsEBB && not (canBeEBB actualSlotNo)) $
         throwError $ UnexpectedEBBInSlot actualSlotNo
+      unless (fromIsEBB newIsEBB) $ -- TODO fine to ignore EBBs?
+        whenJust (Map.lookup (blockNo hdr) checkpoints) $ \checkpoint ->
+          unless (checkpoint == blockHash hdr) $
```

Comment:
Minor: I think the prevailing idiom is to prefer `when` over `unless` when the guard logic is easy to read either way.

Reply:
https://github.com/input-output-hk/ouroboros-consensus/pull/453#discussion_r1371394977

Reply:
Switched unless to when in 5c93fa0